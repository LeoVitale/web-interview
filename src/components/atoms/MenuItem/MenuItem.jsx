import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Icon from 'components/atoms/Icon'
import { menuItem, iconItem } from './MenuItem.module.scss'

const MenuItem = ({ className, icon, label, to }) => {
  return (
    <Link className={`${menuItem} ${className}`} to={to}>
      <Icon type={icon} className={iconItem} />
      <span>{label}</span>
    </Link>
  )
}

MenuItem.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
}

MenuItem.defaultProps = {
  className: '',
  icon: '',
}

export default MenuItem
