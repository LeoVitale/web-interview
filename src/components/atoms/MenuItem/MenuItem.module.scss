@import 'styles/_constants.scss';

.menuItem {
  text-decoration: none;
  font-size: 1.6em;
  color: $bold-text-color;
  margin-right: 25px;
}
