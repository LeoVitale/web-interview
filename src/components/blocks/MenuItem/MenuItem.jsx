import React from 'react';
import { Link } from 'react-router-dom';
import styles from './MenuItem.module.scss';

const MenuItem = ({ icon, label, to }) => {
  const { menuItem } = styles;
  return (
    <Link className={menuItem} to={to}>
      <i>{icon}</i>
      <span>{label}</span>
    </Link>
  );
};

export default MenuItem;
