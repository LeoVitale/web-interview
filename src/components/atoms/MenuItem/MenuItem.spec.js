import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount } from 'enzyme'
import MenuItem from './MenuItem.jsx'

describe('<MenuItem />', () => {
  const menuItem = mount(
    <Router>
      <MenuItem to="" />
    </Router>
  )

  it('should render MenuItem as expected', () => {
    expect(menuItem).toMatchSnapshot()
  })
})
