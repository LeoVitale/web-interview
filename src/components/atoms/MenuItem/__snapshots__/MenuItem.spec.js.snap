// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> should render MenuItem as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MenuItem
      to=""
    >
      <Link
        className="menuItem"
        replace={false}
        to=""
      >
        <a
          className="menuItem"
          href="/"
          onClick={[Function]}
        >
          <Icon>
            <svg
              className="svgIcon"
              fill="#400099"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <title />
              <path
                d="M19.504 17.997h1.75a.75.75 0 0 1 0 1.5h-1.75v1.75a.75.75 0 1 1-1.5 0v-1.75h-1.75a.75.75 0 1 1 0-1.5h1.75v-1.75a.75.75 0 0 1 1.5 0v1.75zM3.527 8.25h17.05V5a.25.25 0 0 0-.25-.25h-3.32v1.092a.75.75 0 1 1-1.5 0V4.75H8.441v1.038a.75.75 0 0 1-1.5 0V4.75H3.777a.25.25 0 0 0-.25.25v3.25zm17.05 1.5H3.527V19c0 .138.112.25.25.25h9.526a.75.75 0 1 1 0 1.5H3.777A1.75 1.75 0 0 1 2.027 19V5c0-.966.784-1.75 1.75-1.75h3.164v-.449a.75.75 0 1 1 1.5 0v.449h7.066v-.449a.75.75 0 1 1 1.5 0v.449h3.32c.966 0 1.75.784 1.75 1.75v9.601a.75.75 0 1 1-1.5 0V9.75z"
              />
            </svg>
          </Icon>
          <span />
        </a>
      </Link>
    </MenuItem>
  </Router>
</BrowserRouter>
`;
